🌱 feat/page-members-filters

walidkorchi opened the pull request:
> ⚠️ This PR will most likely receive updates if one the tasks below are not completed yet.

## Features

- [x] add internationalization to website with 4 supported languages (fr, en, es, zh)
- [x] Replace both linter & formatter respectively from eslint/prettier to Biome toolchain
- [x] Filters supports generic types and is perfectly reusable across web app
- [x] News (Actualités) page with MDC filesystem as strategy for posting articles in news, each file under `/news/` has a required param image attached to it under `/public/news` having the same name of the markdown file for the sake of conformity
- [x] New internal design system with inline theming using Reka UI's unstyled and highly accessible primitives with the imported components: `Avatar` `Badge` `Breadcrumb` `Button` `Card` `Input` `Resizable` `ScrollArea` `Select` `Skeleton` `Tooltip`
~~- [ ] Dynamic translation for modifiers (Pages > Scénarios) descriptions~~ Removed as deprecated page

## Fixes

- [x] Removed 2 unused vue components in codebase
- [x] Removed twitter from seo open graph meta tags as no longer supported social platform
- [ ] Exported vue component `ContentRenderer` from `@nuxt/content` does not seem to support HMR in a dev environment when modifying a markdown file, a full server reload is required and not good for a DX
- [ ] Add custom global-error page (404/500) right before nuxt throws its UI that does not match the project's color schema

## Website visuals

- [x] Add language selector to header as a dropdown menu
- [x] Improve design of `Nav.vue` as a floating compact navigation bar and with backdrop blur filter
- [x] Add a copy IP nav button displaying the count of connected players to encourage people joining
- [x] Wrap loading images with skeletons in the mean time
- [x] Improve layout of members page by compacting cards from 3 grid cols to 4 while keeping good spacing and less scrolling
- [x] Display player skin avatar in between logs messages to highlight attackers, deaths, kills actions in games page
- [x] Implement sticky search input to filter archived games in aside tag of games page
- [x] Implement resizable panels and width-based dynamic display of player skin's avatar

## Optimizations & Refactors

- [x] Reduce nodes from DOM by substituting elements
- [x] Consolidate SEO constants into single source of truth, same with API types + endpoint

## What's Next ?

- Remove `stats.modifiers` property from API as no longer used by the front-end
- Replace Inter font family with Monocraft, could highly improve the Minecraft aspect

walidkorchi committed:
feat(members/filteres): add generic types

walidkorchi committed:
feat(lint/format): replace prettier/eslint with biome toolchain

walidkorchi committed:
refactor(i18n): add translation
 + optimize & clean up

walidkorchi committed:
refactor: clean up + update docs

walidkorchi committed:
refactor(i18n): add spanish + optimize DOM

Romitou reviewed (approved):
LGTM, merci :)
Closes #46

walidkorchi committed:
fix(pages/modifies+assets): remove deprecated page + static assets

walidkorchi committed:
feat(design): new design system + out of the box tw animations

walidkorchi committed:
feat(components): new tooltip matching design system

walidkorchi committed:
feat(components): new skeleton for loading states UIs

walidkorchi committed:
feat(components): new select for styled item selection

walidkorchi committed:
feat(components): cross-browser scrolling augmenting native scroll

walidkorchi committed:
feat(components): new resizable panel groups + layouts with keyboard
support

walidkorchi committed:
feat(components): new input versatile component

walidkorchi committed:
feat(components): new card with header, content, and footer supports

walidkorchi committed:
feat(components): new button component with high accessibility

walidkorchi committed:
feat(components): new breadcrumb displaying hierarchy of links

walidkorchi committed:
feat(components): new badge component to render tags

walidkorchi committed:
feat(components): new image element with a fallback support

walidkorchi committed:
feat(pages): add mdc file structure for news page

walidkorchi committed:
refactor(seo): introduce chinese locale + clean up

walidkorchi committed:
refactor(config): necessary code clean up

walidkorchi committed:
refactor: many UI improvements with inline theming

- leverage design system with the new components powered by reka-ui
- add skin face player in between logs messages to highlight the actions
  of the archived game
- substitute css webkit scrollbar styling with accessible scroll-area
  component
- remove deprecated modifiers code (still needs to be removed from API)
- fetch count of players connected in the server to encourage people
  joining the server
- improve members's filters

walidkorchi committed:
refactor(news): inter image post as required param to be found under
public/news + use inline theme + format date locale-based

walidkorchi committed:
refactor: update lock file

walidkorchi committed:
fix(seo+news): missing logo in meta tags + path url resolve + add
theme-color

walidkorchi committed:
feat(seo/og-image): try out takumi framework agnostic OG image gen
